edx_solutions_api_integration/users/views.py

```diff
@@ -759,25 +760,16 @@ def post(self, request, user_id):  # pylint: disable=R0915
                 new_profile_image_names = get_profile_image_names(username)
                 for old_image_size, old_image_name in profile_image_names.items():
                     if storage.exists(old_image_name):
-                        old_image_path = os.path.join(storage.location, old_image_name)
-                        new_image_path = os.path.join(storage.location, new_profile_image_names[old_image_size])
                         try:
-                            os.rename(old_image_path, new_image_path)
-                        except OSError:
-                            raise
-
-            if old_username != username:
-                storage = get_profile_image_storage()
-                profile_image_names = get_profile_image_names(old_username)
-                new_profile_image_names = get_profile_image_names(username)
-                for old_image_size, old_image_name in profile_image_names.items():
-                    if storage.exists(old_image_name):
-                        old_image_path = storage.path(old_image_name)
-                        new_image_path = os.path.join(storage.location, new_profile_image_names[old_image_size])
-                        try:
-                            os.rename(old_image_path, new_image_path)
+                            file = storage.open(old_image_name)
+                            storage.save(new_profile_image_names[old_image_size], ContentFile(file.read()))
                         except OSError:
```

Comment:
an `IOError` is raised if there's issue with file opening

Reply:
done